File: frontend/src/app/pacientes/components/formulario-paciente/models/cep.ts
export class CEP {
    id: number;
    cep: string;
    logradouro: string;
    bairro: string;
    municipioId: number;
    ufId: number;

    constructor(
        id: number,
        cep: string,
        logradouro: string,
        bairro: string,
        municipioId: number,
        ufId: number,
    ) {
        this.id = id;
        this.cep = cep;
        this.logradouro = logradouro;
        this.bairro = bairro;
        this.municipioId = municipioId;
        this.ufId = ufId;
    }
}

File: pacientes/src/main/java/br/com/basis/madre/domain/CEP.java
package br.com.basis.madre.domain;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;

import javax.persistence.*;

import org.springframework.data.elasticsearch.annotations.Field;
import org.springframework.data.elasticsearch.annotations.FieldType;
import java.io.Serializable;

/**
 * A CEP.
 */
@Data
@NoArgsConstructor
@Entity
@Table(name = "cep")
@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
@org.springframework.data.elasticsearch.annotations.Document(indexName = "cep")
public class CEP implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "sequenceGenerator")
    @SequenceGenerator(name = "sequenceGenerator")
    @org.springframework.data.elasticsearch.annotations.Field(type = FieldType.Keyword)
    private Long id;

    @Column(name = "cep")
    private String cep;

    @Column(name = "logradouro")
    private String logradouro;

    @Column(name = "bairro")
    private String bairro;


    @ManyToOne
    @JoinColumn(name = "municipio_id", referencedColumnName = "id")
    private Municipio municipio;

    @ManyToOne
    @JoinColumn(name = "uf_id", referencedColumnName = "id")
    private UF uf;

    public CEP cep(String cep) {
        this.cep = cep;
        return this;
    }

    public CEP logradouro(String logradouro) {
        this.logradouro = logradouro;
        return this;
    }

    public CEP bairro(String bairro) {
        this.bairro = bairro;
        return this;
    }

}
